containerd-shim-v2/service.go

```diff
@@ -274,7 +302,20 @@ func getTopic(e interface{}) string {
 	return cdruntime.TaskUnknownTopic
 }
 
+func trace(ctx context.Context, name string) opentracing.Span {
+	if ctx == nil {
+		logrus.WithField("type", "bug").Error("trace called before context set")
+		ctx = context.Background()
+	}
+	span, _ := opentracing.StartSpanFromContext(ctx, name)
```

Comment:
Shoudnt you be fetching the context returned from StartSpanFromContext here? This function should return the context which should be used henceforth in the calling function.